x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -191,10 +192,26 @@ export function trainedModelsRoutes(
             mlLog.debug(e);
           }
 
-          const body = filterForEnabledFeatureModels(result, getEnabledFeatures());
+          try {
+            if (withOriginJobCheck) {
+              for (const model of result) {
```

Comment:
This loop could be large and it's calling `getDataFrameAnalytics` each time.
`getDataFrameAnalytics` can take a list of IDs, so this could be rewritten to perform one call to `getDataFrameAnalytics` but with a list of all the IDs which are DFA jobs.

Reply:
Updated to single call, bypassing 404 for missing jobs with wildcard added to each job id and `allow_no_match` set to true. 7c06abd24a23cdfb98b34fba27bfdce6aee3e1ad